### DISCUSSION TITLE
Cannot activate face recognition

### DISCUSSION THREAD

**Discussion by peterpedron** [photoprism/photoprism#4402] [Q&A]
#### 1. What is not working as documented?

When activating the "People" featur in the settings, the API seems to be called, a menu entry on the left side is visibl for half a second, but eventually the function switches back to off.
![image](https://github.com/photoprism/photoprism/assets/3693909/c66c13df-258a-43db-b3da-9e22a5c617a6)

#### 2. How can we reproduce it?

1. Use raspberrypi4 armv7 docker image on raspberrypi4
2. Activate "people" function

I have 110 images in my gallery up to now.

#### 3. What behavior do you expect?

People function switched on

#### 4. What could be the cause of your problem?

No idea, all configs seem to be fine.

#### 5. Can you provide us with example files for testing, error logs, or screenshots?

No entry in debug log, other than
```
photoprism-1  | time="2024-07-02T21:52:29Z" level=trace msg="config: cache hit for wallpaper-uri"
photoprism-1  | time="2024-07-02T21:52:29Z" level=trace msg="config: updated counts [79.451793ms]"
photoprism-1  | time="2024-07-02T21:52:29Z" level=debug msg="server: POST /api/v1/settings (200) [696.829997ms]"
```

log level set to "trace" in settings.

#### 6. Which software versions do you use?

(a) PhotoPrism Architecture & Build Number: 

Build [240531-60b3a4628](https://docs.photoprism.app/release-notes/), ARMv7,...

(b) Database Type & Version: MariaDB,

(c) Operating System Types & Versions: Linux, buster.

(d) Browser Types & Versions:  Chrome

(e) Ad Blockers, Browser Plugins, and/or Firewall Software? no

You can find the version/build number of the app in *Settings* by scrolling to the bottom. Note that MySQL 8 support has been discontinued, see system requirements at <https://docs.photoprism.app/getting-started/#system-requirements>.

*Always provide database and operating system details if it is a backend, import, or indexing issue. Should it be a frontend issue, at a minimum we require you to provide web browser and operating system details. When reporting a performance problem, we ask that you provide us with complete information about your environment, as there may be more than one cause.*

#### 7. On what kind of device is PhotoPrism installed?

This is especially important if you are reporting a performance, import, or indexing issue. You can skip this if you're reporting a problem you found in our public demo, or if it's a completely unrelated issue, such as incorrect page layout.

(a) Device / Processor Type: Raspberry Pi 4, Intel Core i7-3770, AMD Ryzen 7 3800X,...

(b) Physical Memory & Swap Space in GB

(c) Storage Type: HDD, SSD, RAID, USB, Network Storage,...

(d) Anything else that might be helpful to know?

*Always provide device, memory, and storage details if you have a backend, performance, import, or indexing issue.*

#### 8. Do you use a Reverse Proxy, Firewall, VPN, or CDN?

If yes, please specify type and version. You can skip this if you are reporting a completely unrelated issue.

*Always provide this information when you have a reliability, performance, or frontend problem, such as failed uploads, connection errors, broken thumbnails, or video playback issues.*

**Using NGINX?** Please also provide the configuration and/or consider asking the NGINX community for advice as we do not specialize in supporting their product. Docs can be found at <https://docs.photoprism.app/getting-started/proxies/nginx/>.

**Comment by graciousgrey** [comment]
Is it possible that [tensorflow is disabled](https://docs.photoprism.app/user-guide/settings/advanced/#disable-tensorflow)?

Please note that RAW image conversion and TensorFlow are disabled on systems with 1 GB or less memory.
- https://docs.photoprism.app/getting-started/raspberry-pi/#system-requirements

**Comment by peterpedron** [comment]
Hi,
my raspbi has 4GB RAM (see screenshot from htop)
![image](https://github.com/photoprism/photoprism/assets/3693909/9c13016b-1bf6-4f11-97c2-6b8544824f67)

Also, I could not see Tensorflow being disabled with `PHOTOPRISM_DISABLE_TENSORFLOW`
![image](https://github.com/photoprism/photoprism/assets/3693909/b0b0bbcc-d1eb-41bd-aa2a-94aec8826a5d)

In the `docker-compose.yml`, the option `PHOTOPRISM_DISABLE_TENSORFLOW` is also not explicitely mentioned.

Image classification seems to work though...
See here:
![image](https://github.com/photoprism/photoprism/assets/3693909/91dd7c7a-4059-4739-a14e-165fdb9a6d10)

**Comment by graciousgrey** [comment]
Ok, so it seems tensorflow is enabled and working :+1: 

Maybe PHOTOPRISM_DISABLE_FACES is set to true in your docker-compose.yml ?
